src/Microsoft.AspNetCore.SignalR.Client/HubConnectionBuilder.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.Net.Http;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Microsoft.AspNetCore.Sockets;
+using Microsoft.AspNetCore.Sockets.Client;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    public class HubConnectionBuilder
+    {
+        private readonly Uri _url;
+        private ILoggerFactory _loggerFactory;
+        private TransportType? _transportType;
+        private IHubProtocol _hubProtocol;
+        private HttpMessageHandler _httpMessageHandler;
+
+        public HubConnectionBuilder(Uri url)
+        {
+            _url = url;
+        }
+
+        public HubConnectionBuilder WithLogger(ILoggerFactory loggerFactory)
```

Comment:
This might just be me but "With" doesn't feel like the right word here. Maybe "Add"

Reply:
`With` is used by other fluent builder APIs, like Roslyn's `Compilation` http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/Compilation.cs,308

However, I wonder if `Use` might fit better with our patterns.